xray/segment.go

```diff
@@ -449,15 +477,20 @@ func (seg *Segment) beforeEmitSubsegment(s *Segment) {
 
 // AddAnnotation allows adding an annotation to the segment.
 func (seg *Segment) AddAnnotation(key string, value interface{}) error {
+	seg.Lock()
```

Comment:
why do we lock and unlock before checking dummy?

Reply:
because we don't want any data race on segment struct. so it's a good idea to lock the struct before accessing it.